src/providers/subgraph-provider.ts

```diff
@@ -22,7 +20,41 @@ export interface ISubgraphProvider<TSubgraphPool extends SubgraphPool> {
 
 const PAGE_SIZE = 1000; // 1k is max possible query size from subgraph.
 
-export abstract class SubgraphProvider {
+export type V3V4SubgraphPool = {
+  id: string;
+  feeTier: string;
+  liquidity: string;
+  token0: {
+    id: string;
+  };
+  token1: {
+    id: string;
+  };
+  tvlETH: number;
+  tvlUSD: number;
+};
+
+export type V3V4RawSubgraphPool = {
+  id: string;
+  feeTier: string;
+  liquidity: string;
+  token0: {
+    symbol: string;
+    id: string;
+  };
+  token1: {
+    symbol: string;
+    id: string;
+  };
+  totalValueLockedUSD: string;
+  totalValueLockedETH: string;
+  totalValueLockedUSDUntracked: string;
+};
+
+export abstract class SubgraphProvider<
+  TRawSubgraphPool extends V3V4RawSubgraphPool,
+  TSubgraphPool extends V3V4SubgraphPool
```

Comment:
got a bit confused with `V3V4SubgraphPool` vs `V3SubgraphPool`,`V3SubgraphPool` (those are still defined in v3/&v4/ subgraph-provider.ts). so here you are saying that both should have those common fields?

Reply:
yep, v3 v4 have a lot of common fields, meanwhile v2 is not the case.

Reply:
sadly its kinda confusing. because v3 v4 have concentrated liquidities, so v3 v4 pools data structure look very similar, meanwhile v2 pool data structure is entirely different.

In future, I see us stay in concentrated liquidities, hence I see the effort to refactor v2 subgraph provider is not best spent.